### PR TITLE
feat: replace traefik with istio-ingress for public ingress

### PR DESCRIPTION
**Pull request by wood-push-melon**

### CODE REVIEW

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -2,72 +2,116 @@
...
+    # Test HTTP to HTTPS redirection
+    http_endpoint = (
+        f"http://{public_ingress_address}/{ops_test.model_name}-{LOGIN_UI_APP}/ui/login"
+    )
+    resp = await http_client.get(
+        http_endpoint,
+        headers={"Host": PUBLIC_INGRESS_DOMAIN},
+    )
+    assert resp.status_code == http.HTTPStatus.MOVED_PERMANENTLY, (
+        f"Expected HTTP 301 for {http_endpoint}, got {resp.status_code}."
```
Comment (pik4ez-canonical) [inline tests/integration/test_charm.py]: Here a hardcoded value `301` is used. On the line before the same thing expressed via `http.HTTPStatus.MOVED_PERMANENTLY` constant. Does it make sense to stick to one of the options?

**Review by pik4ez-canonical — DISMISSED** [review]
One minor suggestion. Overall LGTM.

File: tests/integration/test_charm.py (hunk 1 above)
Comment (wood-push-melon) [reply tests/integration/test_charm.py]: IMHO, the `MOVED_PERMANENTLY` is for people who program the tests. The `301` in the assert failure message is for whoever runs the tests. So this way it should be readable for both.

Comment (wood-push-melon) [reply tests/integration/test_charm.py]: Oh, my bad. I understand what u mean. I can change it.

**Review by pik4ez-canonical — APPROVED** [review]

**Pull request merged**